Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  CardNumberViewModel.swift
+//  Frames
+//
+//  Created by Harry Brown on 07/07/2022.
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import Checkout
+
+protocol CardNumberViewModelDelegate: AnyObject {
+  func update(cardNumber: String, scheme: Card.Scheme)
+}
+
+protocol CardNumberViewModelProtocol {
+  func textFieldUpdate(from: String) -> String?
+}
+
+class CardNumberViewModel {
+  weak var delegate: CardNumberViewModelDelegate?
+  weak var cardNumberView: CardNumberViewProtocol?
```

Comment:
I think this UI view interface could be merged with `CardNumberViewModelDelegate` and removed.

```swift
protocol CardNumberViewModelDelegate: AnyObject {
   func viewModelUpdate(cardNumber: String, scheme: Card.Scheme)
   func viewModelUpdate(schemeIcon: Constants.Bundle.SchemeIcon)
 }

// Then 
cardNumberView?.schemeIcon = .init(scheme: scheme)
// Becomes
delegate.viewModelUpdated(schemeIcon: .init(scheme: scheme)
```

Currently your view owns viewmodel and viewmodel owns the view twice, once as a delegate and again as a view interface. Implementation is safe now (not memory retain cycle), but that duplication of ownership could be merged into 1

Reply:
As I look more, I also wonder whether you want to merge the functions as well, I think they are always called together anyway

Reply:
will address in a follow up pr as part of error checking. please approve

Reply:
Sure. If you have the approvals you can always merge without my explicit approval, I'm happy to follow up later or to have your opinion regardless 👍 . Will not take it personal 😛